orangecontrib/network/widgets/OWNxAnalysis.py

```diff
@@ -246,8 +250,28 @@ def set_graph(self, graph):
         self.cancel_job()
         self.graph = graph
 
+        allowed_edge_types = {GENERAL, DIRECTED, UNDIRECTED}
+        if graph is not None:
+            allowed_edge_types.remove(UNDIRECTED if self.graph.edges[0].directed else DIRECTED)
+
         self.known = {}
         for name in METHODS:
+            curr_method = METHODS[name]
+            if curr_method.level == INTERNAL:
+                continue
+
+            lbl_obj = getattr(self, "tool_{}".format(name))
+            cb_obj = self.method_cbs[name]
+
+            # disable/re-enable valid graph indices
+            if curr_method.edge_constraint not in allowed_edge_types:
+                lbl_obj.setDisabled(True)
+                cb_obj.setChecked(False)
```

Comment:
We usually keep the checkbox state when disabling it, so that it is "restored" toe the same state when re-enabled.

We can keep this code as it is, but modify it when checkboxes with specific state when disabled (https://github.com/biolab/orange-widget-base/issues/24) are implemented.